### DISCUSSION TITLE
Architecture / Integration Diagram

### DISCUSSION THREAD

**Discussion by abitrolly** [nasa/openmct#7346] [Community Q&A] [answered]
As a former DevOps / Automation Engineer, who has little to no JavaScript background, I'd like to ask for a diagram that depicts OpenMCT as a deployment object. Over the first look documentation about OpenMCT structure is `git` and JavaScript oriented. It would be nice to get a detailed perspective from a different angle. In particular, it the diagram could answer the questions.

1. What OpenMCT deployment consists of?
2. How does OpenMCT fits in a complete telemetry system with senders, archive, and storage components?
3. Which network protocols are used out of the box?
4. What is protocol direction? Is it OpenMCT that establishes connection to devices and needs to know device auth params? Or is it device that somehow pushes data to OpenMCT server with the given keys?
5. How the connection loss is handed/
6. Who and where logs connection errors, value errors (unexpected data) and abnormal telemetry (out of range of standard operation)?

Given the page https://nasa.github.io/openmct/whos-using-open-mct/ it is useful to have the integration diagram for each project to answer these questions. Maybe planning a podcast series for OpenMCT in the next mission will help gather time to cover that.

**Comment by unlikelyzero** [comment, marked as answer]
@abitrolly https://github.com/scottbell/openmct-quickstart?tab=readme-ov-file#diagram
